examples/vecbuiltin/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+#=============================================================================
+#   CMake build system files
+#
+#   Copyright (c) 2024 Michal Babej / Intel Finland oy
+#
+#   Permission is hereby granted, free of charge, to any person obtaining a copy
+#   of this software and associated documentation files (the "Software"), to deal
+#   in the Software without restriction, including without limitation the rights
+#   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#   copies of the Software, and to permit persons to whom the Software is
+#   furnished to do so, subject to the following conditions:
+#
+#   The above copyright notice and this permission notice shall be included in
+#   all copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#   THE SOFTWARE.
+#
+#=============================================================================
+
+if(NOT ENABLE_HOST_CPU_VECTORIZE_BUILTINS)
+  return()
+endif()
+
+add_compile_options(${OPENCL_CFLAGS})
+
+add_executable("vecbuiltin" vecbuiltin.c vecbuiltin_exec.c vecbuiltin.cl)
+target_link_libraries("vecbuiltin" ${POCLU_LINK_OPTIONS})
+add_symlink_to_built_opencl_dynlib("vecbuiltin")
+
+# the llvm-filecheck only works with LLVM 19
+if(LLVM_VERSION VERSION_GREATER_EQUAL 19.0)
+  add_test_pocl(NAME "examples/vecbuiltin" COMMAND "vecbuiltin" LLVM_FILECHECK "vecbuiltin.fc" WORKITEM_HANDLER "loopvec")
+else()
+  add_test_pocl(NAME "examples/vecbuiltin" COMMAND "vecbuiltin" WORKITEM_HANDLER "loopvec")
```

Comment:
Maybe this test is a bit pointless if FC doesn't work. The check for filecheck is inside the add_test_pocl() ?

Reply:
Yeah probably pointless, i'll remove the else branch.

The "working" filecheck is not the problem; the problem is that different versions of LLVM vectorize different set of builtins. The current `example/vecbuiltin` doesn't get vectorized at all by older LLVMs, so inevitably filecheck reports a failure. So....  i'd have to prepare a separate version of the test for each LLVM version, or have a single version with so little builtins that it's (hopefully) vectorized by all LLVMs. I didn't see either option as preferable, so instead i've limited the filecheck test to LLVM 19.

Reply:
OK. Makes sense. We should not put any effort to the older LLVMs vectorization-wise.